types/state/chain_manager.go

```diff
@@ -73,7 +73,10 @@ func (c *ChainManager) Entries(start int64, end int64) ([][]byte, error) {
 	// multiple times
 	entries := make([][]byte, 0, end-start)
 	for start < end {
-		h, err := c.merkle.GetRange(c.key, start, end)
+		if err := c.merkle.SetKey(c.key...); err != nil {
+			return nil, err
+		}
```

Comment:
`ChainManager` holds its own copy of the merkle manager with the key set, so this is unnecessary.